### PR TITLE
DIT-4881 Make EORI editable depending on CDS Flag

### PR DESCRIPTION
**Pull request by ed0906**
I've implemented it as follows:
- The existing CDS bypass flag in config decides wether its permitted for someone to not have an EORI

After this point the flag isn't used and everything hangs off whether the EORI is present on the request, which has involved:

- Making the Request EORI optional
- Making the main journey toggle content based on whether the request EORI is present

### CODE REVIEW

**Comment by ed0906** [comment]
Dont merge until It's been tested etc :)

**Comment by ed0906** [comment]
@jumpertester The IDs on the Register Business For EORI page may have changed (from `field1`, `field2`) to something more meaningful e.g. `businessName`, `addressLine1`. I imagine this will cause a few test failures

File: conf/application.conf [hunk 1]
```diff
@@ -169,7 +169,7 @@ filters {
   }
 }
 
-cdsEnrolmentCheckEnabled = true
+cdsEnrolmentCheckEnabled = false
```
Comment (ed0906) [inline conf/application.conf]: This is overridden in production

File: app/models/WhichBestDescribesYou.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ object WhichBestDescribesYou {
     }
   }
 
-  def isBusinessRepresentative(answers: UserAnswers): Boolean = {
+  def theUserIsAnAgent(answers: UserAnswers): Boolean = {
```
Comment (gokyo) [inline app/models/WhichBestDescribesYou.scala]: `isSignedInUserAnAgent`

File: app/mapper/CaseRequestMapper.scala [hunk 3]
```diff
@@ -105,15 +99,15 @@ class CaseRequestMapper {
...
 
-  private def buildEoriDetailsFromCurrentUser(eoriNumber: String): RegisteredAddressForEori => EORIDetails = { details: RegisteredAddressForEori =>
+  private def toEoriDetails(details: RegisteredAddressForEori): EORIDetails = {
     EORIDetails(
-      eoriNumber,
-      details.field1, // business name
-      details.field2, // address line 1
-      details.field3, // address line 2 (town)
+      details.eori,
+      details.businessName, // business name
```
Comment (gokyo) [inline app/mapper/CaseRequestMapper.scala]: The comments were useful before, when using field1, field2, etc
Now that you named the fields with the correct names, the comments can go away

File: app/models/WhichBestDescribesYou.scala (hunk 2 above)
Comment (ed0906) [reply app/models/WhichBestDescribesYou.scala]: `if(is...)` never reads very well in my opinion

File: app/mapper/CaseRequestMapper.scala (hunk 3 above)
Comment (ed0906) [reply app/mapper/CaseRequestMapper.scala]: Removed

**Comment by ed0906** [comment]
Merged as part of #87

**Pull request closed without merging**